server/controllers/auth.js

```diff
@@ -114,3 +114,45 @@ exports.logoutUser = asyncHandler(async (req, res, next) => {
 
   res.send("You have successfully logged out");
 });
+
+// @route GET /auth/demo
+// @desc Login user
+// @access Public
+exports.loginDemoUser = asyncHandler(async (req, res, next) => {
+  const email = process.env.DEMO_EMAIL;
+  const username = process.env.DEMO_USERNAME;
+  const password = process.env.DEMO_PASSWORD;
+
+  let user = await User.findOne({ email });
+
+  if (!user) {
+    user = await User.create({
+      email,
+      username,
+      password,
+    });
+  }
```

Comment:
You're supposed to return a `404` if no user exists with this email. A separate route is provided for registering a new user, so shouldn't be done here.

Reply:
Hey, the reason behind this is cause I am implementing the demo user feature and the demo user is based of the .env file. So if the demo user exists log him in, if not create one automatically. This allows for a seamless experience.